besu/src/main/java/org/hyperledger/besu/cli/BesuCommand.java

```diff
@@ -1946,15 +1962,16 @@ private Path pluginsDir() {
     }
   }
 
-  public File nodePrivateKeyFile() {
-    File nodePrivateKeyFile = null;
-    if (isFullInstantiation()) {
-      nodePrivateKeyFile = standaloneCommands.nodePrivateKeyFile;
-    }
+  private NodeKey buildNodeKey() {
+    addConfigurationService(); // its a no-op if besu configuration service is already initialized
+    return new NodeKey(securityModuleProvider(securityModuleProviderName));
+  }
 
-    return nodePrivateKeyFile != null
-        ? nodePrivateKeyFile
-        : KeyPairUtil.getDefaultKeyFile(dataDir());
+  private SecurityModule securityModuleProvider(final String name) {
```

Comment:
probably don't need to pass in the string as a parameter - you can take it from the class variable.

Reply:
👍